taipy/core/_entity/submittable.py

```diff
@@ -86,7 +88,10 @@ def is_ready_to_run(self) -> bool:
         Returns:
             True if the given entity is ready to be run. False otherwise.
         """
-        return all(dn.is_ready_for_reading for dn in self.get_inputs())
+        if self._submittable_id not in SubmittableStatusCache._submittable_id_datanodes:
+            for dn in self.get_inputs():
+                SubmittableStatusCache._compute_if_dn_is_ready_for_reading(dn)
+        return SubmittableStatusCache._check_submittable_is_ready_to_submit(self._submittable_id)
```

Comment:
This does not work since calling `SubmittableStatusCache._compute_if_dn_is_ready_for_reading(dn)` triggers a notification if the scenario is submittable, while the value may not have changed recently.

In any case, why do you need to both access the cache and recompute the value?

I would simply rollback this part to use the previous code using `dn.is_ready_for_reading` directly, which should return the reasons as well.

Reply:
but the `is_ready_for_reading` in DN doesn't update the ReadyToRunCache and only checking the value of the data node? I don't think it will work no?

Reply:
but the `is_ready_for_reading` in DN doesn't update the ReadyToRunCache and only checking the value of the data node? I don't think it will work no?